File: backend/src/main/java/io/ntf/api/infrastructure/jwt/JwtReactiveAuthenticationEntryPoint.java
package io.ntf.api.infrastructure.jwt;

import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.web.server.ServerAuthenticationEntryPoint;
import org.springframework.web.server.ServerWebExchange;
import reactor.core.publisher.Mono;

@Slf4j
public class JwtReactiveAuthenticationEntryPoint implements ServerAuthenticationEntryPoint {
  @Override
  public Mono<Void> commence(ServerWebExchange exchange, AuthenticationException e) {
    exchange.getResponse().setStatusCode(HttpStatus.UNAUTHORIZED);
    return Mono.empty();
  }
}

File: backend/src/main/kotlin/io/ntf/api/infrastructure/jwt/JwtReactiveAuthenticationEntryPoint.kt
package io.ntf.api.infrastructure.jwt

import org.springframework.http.HttpStatus
import org.springframework.security.core.AuthenticationException
import org.springframework.security.web.server.ServerAuthenticationEntryPoint
import org.springframework.web.server.ServerWebExchange
import reactor.core.publisher.Mono

class JwtReactiveAuthenticationEntryPoint : ServerAuthenticationEntryPoint {
  override fun commence(exchange: ServerWebExchange, e: AuthenticationException): Mono<Void> {
    exchange.response.statusCode = HttpStatus.UNAUTHORIZED
    return Mono.empty()
  }
}
